solution/index.js

```diff
@@ -0,0 +1,349 @@
+                    localStorageArray = localStorageObject["todo"]
+                    break;
+                case "done-tasks":
+                    localStorageArray = localStorageObject["done"]
+                    break;
+            }
+            localStorageArray.unshift(valueOfDraggingElement)
+            localStorage.tasks = JSON.stringify(localStorageObject)
+            break;
+        case "removezone":
+            document.querySelector(`.${previousParentClassName}`).removeChild(draggingElement);
+            removeReplacingElement(valueOfDraggingElement);
+
+            if(importantTasksArray.includes(valueOfDraggingElement)){ //Deleting from importantTasksArray deleted tasks
+                importantTasksArray.splice(importantTasksArray.indexOf(valueOfDraggingElement), 1)
+                localStorage.important = JSON.stringify(importantTasksArray)
+            }
+
+            localStorage.tasks = JSON.stringify(localStorageObject)
+            break;
+    }
+}
+
+//Event listener of click on important icon that use toggle(to add or to remove indication)
+function toggleToImportantTasks(event){
+    const target = event.target;
+    const task = target.closest("li").firstChild.innerText
+    if(target.closest("li").className.includes("important")){
+        target.closest("li").classList.remove("important")
+        importantTasksArray.splice(importantTasksArray.indexOf(task), 1);
+        localStorage.important = JSON.stringify(importantTasksArray)
+    } else {
+        target.closest("li").classList.add("important")
+        importantTasksArray.unshift(task)
+        localStorage.important = JSON.stringify(importantTasksArray)
+    }
+    
+}
+
+//Function that check all existing tasks if them important tasks
+function generationImportantTasksFromLocalStorage(importantTasksArray){
+    const allTasksElements = document.querySelectorAll("li")
+    for(const task of allTasksElements){
+        for(const importantTask of importantTasksArray){
+            if(task.firstChild.innerText === importantTask){
+                task.classList.add("important")
+            }
+        }
+    }
+}
+
+
+let draggingElement;
+
+const loaderElement = document.querySelector("#loader")
+
+let localStorageObject = {
+    "todo":[],
+    "in-progress":[],
+    "done":[],
+}
+let importantTasksArray = [];
+
+if(localStorage.getItem("tasks") === null){
+    localStorage.setItem("important","")
+    localStorage.setItem("tasks","");
+    localStorage.important = JSON.stringify(importantTasksArray)
+    localStorage.tasks = JSON.stringify(localStorageObject);
+} else {
+    importantTasksArray = JSON.parse(localStorage.getItem("important"));
+    localStorageObject = JSON.parse(localStorage.getItem("tasks"));
+    generationTasklistFromLocalStorage(localStorageObject);
+    generationImportantTasksFromLocalStorage(importantTasksArray)
+}
+
+document.querySelector("main").addEventListener("click", addTaskClickEvent)
```

Comment:
Adding an event to the document level is a wrong way of thinking. An event must be created specifically